diff -uNr qa_test_coreutils/orig_test_suite/cp/fail-perm new/orig_test_suite/cp/fail-perm
--- qa_test_coreutils/orig_test_suite/cp/fail-perm	2013-09-22 14:33:17.424171290 +0800
+++ new/orig_test_suite/cp/fail-perm	2014-03-31 16:11:40.553223881 +0800
@@ -42,7 +42,7 @@
 ln -s D/D symlink
 touch F
 cat > exp <<\EOF
-cp: accessing `symlink': Permission denied
+cp: failed to access 'symlink': Permission denied
 EOF
 
 cp F symlink 2> out && fail=1
diff -uNr qa_test_coreutils/orig_test_suite/cp/into-self new/orig_test_suite/cp/into-self
--- qa_test_coreutils/orig_test_suite/cp/into-self	2013-09-22 14:33:17.424171290 +0800
+++ new/orig_test_suite/cp/into-self	2014-03-31 16:11:40.553223881 +0800
@@ -36,7 +36,7 @@
 cp -R dir dir 2> out && fail=1
 
 cat > exp <<\EOF
-cp: cannot copy a directory, `dir', into itself, `dir/dir'
+cp: cannot copy a directory, 'dir', into itself, 'dir/dir'
 EOF
 #'
 
diff -uNr qa_test_coreutils/orig_test_suite/mv/diag new/orig_test_suite/mv/diag
--- qa_test_coreutils/orig_test_suite/mv/diag	2013-09-22 14:33:17.440171291 +0800
+++ new/orig_test_suite/mv/diag	2014-03-31 16:11:40.557223929 +0800
@@ -42,11 +42,11 @@
 
 cat > exp <<\EOF
 mv: missing file operand
-Try `mv --help' for more information.
-mv: missing destination file operand after `no-file'
-Try `mv --help' for more information.
-mv: target `f1' is not a directory
-mv: target `f2' is not a directory
+Try 'mv --help' for more information.
+mv: missing destination file operand after 'no-file'
+Try 'mv --help' for more information.
+mv: target 'f1' is not a directory
+mv: target 'f2' is not a directory
 EOF
 
 cmp out exp || fail=1
diff -uNr qa_test_coreutils/orig_test_suite/mv/dup-source new/orig_test_suite/mv/dup-source
--- qa_test_coreutils/orig_test_suite/mv/dup-source	2013-09-22 14:33:17.440171291 +0800
+++ new/orig_test_suite/mv/dup-source	2014-03-31 16:11:40.557223929 +0800
@@ -49,9 +49,9 @@
   rm -fr a d; touch a; mkdir d
   $i --backup a a d/ 2>> out && fail=1
   cat <<EOF > exp
-$i: warning: source file \`a' specified more than once
-$i: warning: source file \`a' specified more than once
-$i: will not overwrite just-created \`d/a' with \`a'
+$i: warning: source file 'a' specified more than once
+$i: warning: source file 'a' specified more than once
+$i: will not overwrite just-created 'd/a' with 'a'
 EOF
   cmp out exp || fail=1
   test $fail = 1 && diff out exp 2> /dev/null
@@ -65,8 +65,8 @@
   rm -fr a d; touch a; mkdir d
   $i ./a a d/ 2>> out && fail=1
   cat <<EOF > exp
-$i: cannot stat \`a': No such file or directory
-$i: cannot stat \`a': No such file or directory
+$i: cannot stat 'a': No such file or directory
+$i: cannot stat 'a': No such file or directory
 EOF
   cmp out exp || fail=1
   test $fail = 1 && diff out exp 2> /dev/null
diff -uNr qa_test_coreutils/orig_test_suite/mv/force new/orig_test_suite/mv/force
--- qa_test_coreutils/orig_test_suite/mv/force	2013-09-22 14:33:17.440171291 +0800
+++ new/orig_test_suite/mv/force	2014-03-31 16:11:40.557223929 +0800
@@ -28,7 +28,7 @@
 mv $ff $ff > out 2>&1 && fail=1
 
 cat > exp <<EOF
-mv: \`$ff' and \`$ff' are the same file
+mv: '$ff' and '$ff' are the same file
 EOF
 
 cmp out exp || fail=1
diff -uNr qa_test_coreutils/orig_test_suite/mv/perm-1 new/orig_test_suite/mv/perm-1
--- qa_test_coreutils/orig_test_suite/mv/perm-1	2013-09-22 14:33:17.444171291 +0800
+++ new/orig_test_suite/mv/perm-1	2014-03-31 16:11:40.557223929 +0800
@@ -33,7 +33,7 @@
 
 mv no-write/dir . > out 2>&1 && fail=1
 cat <<\EOF > exp
-mv: cannot move `no-write/dir' to `./dir': Permission denied
+mv: cannot move 'no-write/dir' to './dir': Permission denied
 EOF
 
 cmp out exp || fail=1
diff -uNr qa_test_coreutils/orig_test_suite/rm/cycle new/orig_test_suite/rm/cycle
--- qa_test_coreutils/orig_test_suite/rm/cycle	2013-09-22 14:33:17.452171291 +0800
+++ new/orig_test_suite/rm/cycle	2014-03-31 16:11:40.533223640 +0800
@@ -32,8 +32,8 @@
 
 rm -rf a a 2>&1 | sed 's/:[^:]*$//' > out || fail=1
 cat <<\EOF > exp
-rm: cannot remove `a/b/file'
-rm: cannot remove `a/b/file'
+rm: cannot remove 'a/b/file'
+rm: cannot remove 'a/b/file'
 EOF
 
 cmp out exp || fail=1
diff -uNr qa_test_coreutils/orig_test_suite/rm/fail-2eperm new/orig_test_suite/rm/fail-2eperm
--- qa_test_coreutils/orig_test_suite/rm/fail-2eperm	2013-09-22 14:33:17.452171291 +0800
+++ new/orig_test_suite/rm/fail-2eperm	2014-03-31 16:11:40.533223640 +0800
@@ -43,7 +43,7 @@
 sed "s/Not owner/$onp/;s/Permission denied/$onp/" out-t > out
 
 cat <<\EOF > exp
-rm: cannot remove `a/b': Operation not permitted
+rm: cannot remove 'a/b': Operation not permitted
 EOF
 
 cmp out exp || fail=1
diff -uNr qa_test_coreutils/orig_test_suite/rm/inaccessible new/orig_test_suite/rm/inaccessible
--- qa_test_coreutils/orig_test_suite/rm/inaccessible	2013-09-22 14:33:17.452171291 +0800
+++ new/orig_test_suite/rm/inaccessible	2014-03-31 16:11:40.533223640 +0800
@@ -37,7 +37,7 @@
 test -d $p/abs2 && fail=1
 
 cat <<\EOF > exp || fail=1
-rm: cannot remove `rel': Permission denied
+rm: cannot remove 'rel': Permission denied
 EOF
 
 # AIX 4.3.3 fails with a different diagnostic.
diff -uNr qa_test_coreutils/orig_test_suite/rm/rm3 new/orig_test_suite/rm/rm3
--- qa_test_coreutils/orig_test_suite/rm/rm3	2013-09-22 14:33:17.452171291 +0800
+++ new/orig_test_suite/rm/rm3	2014-03-31 16:11:40.533223640 +0800
@@ -61,15 +61,15 @@
 mv o2 out
 
 sort <<EOF > exp || fail=1
-rm: descend into directory \`z'
-rm: remove regular empty file \`z/empty'
-rm: remove write-protected regular file \`z/fu'
-rm: remove write-protected regular empty file \`z/empty-u'
-rm: remove symbolic link \`z/slink'
-rm: remove symbolic link \`z/slinkdot'
-rm: remove directory \`z/d'
-rm: remove write-protected directory \`z/du'
-rm: remove directory \`z'
+rm: descend into directory 'z'
+rm: remove regular empty file 'z/empty'
+rm: remove write-protected regular file 'z/fu'
+rm: remove write-protected regular empty file 'z/empty-u'
+rm: remove symbolic link 'z/slink'
+rm: remove symbolic link 'z/slinkdot'
+rm: remove directory 'z/d'
+rm: remove write-protected directory 'z/du'
+rm: remove directory 'z'
 EOF
 
 cmp out exp || fail=1
diff -uNr qa_test_coreutils/orig_test_suite/rm/rm5 new/orig_test_suite/rm/rm5
--- qa_test_coreutils/orig_test_suite/rm/rm5	2013-09-22 14:33:17.452171291 +0800
+++ new/orig_test_suite/rm/rm5	2014-03-31 16:11:40.533223640 +0800
@@ -28,9 +28,9 @@
 EOF
 
 cat <<\EOF > exp || framework_failure=1
-rm: descend into directory `d'
-rm: remove directory `d/e'
-rm: remove directory `d'
+rm: descend into directory 'd'
+rm: remove directory 'd/e'
+rm: remove directory 'd'
 EOF
 
 if test $framework_failure = 1; then
diff -uNr qa_test_coreutils/orig_test_suite/touch/fail-diag new/orig_test_suite/touch/fail-diag
--- qa_test_coreutils/orig_test_suite/touch/fail-diag	2013-09-22 14:33:17.480171292 +0800
+++ new/orig_test_suite/touch/fail-diag	2014-03-31 16:11:40.533223640 +0800
@@ -35,7 +35,7 @@
 
 touch $dir > out 2>&1 && fail=1
 cat <<EOF > exp
-touch: cannot touch \`$dir': No such file or directory
+touch: cannot touch '$dir': No such file or directory
 EOF
 
 cmp out exp || fail=1
