diff --git a/t/filter/TestFilter/both_str_req_proxy.pm b/t/filter/TestFilter/both_str_req_proxy.pm
index 9e2ba81..4d0b72b 100644
--- a/t/filter/TestFilter/both_str_req_proxy.pm
+++ b/t/filter/TestFilter/both_str_req_proxy.pm
@@ -62,9 +62,7 @@ __DATA__
     <IfModule mod_proxy.c>
         <Proxy http://@servername@:@port@/*>
             <IfModule @ACCESS_MODULE@>
-                Order Deny,Allow
-                Deny from all
-                Allow from @servername@
+                Require host @servername@
             </IfModule>
         </Proxy>
 
diff --git a/t/protocol/TestProtocol/pseudo_http.pm b/t/protocol/TestProtocol/pseudo_http.pm
index bb49ec8..1b8e93b 100644
--- a/t/protocol/TestProtocol/pseudo_http.pm
+++ b/t/protocol/TestProtocol/pseudo_http.pm
@@ -154,9 +154,9 @@ __END__
 
   <Location TestProtocol::pseudo_http>
 
+      <RequireAny>
       <IfModule @ACCESS_MODULE@>
-          Order Deny,Allow
-          Allow from @servername@
+          Require host @servername@
       </IfModule>
 
       <IfModule @AUTH_MODULE@>
@@ -168,7 +168,7 @@ __END__
       AuthName TestProtocol::pseudo_http
       AuthType Basic
       Require user stas
-      Satisfy any
+      </RequireAny>
 
   </Location>
 
diff --git a/t/response/TestAPI/access2.pm b/t/response/TestAPI/access2.pm
index 718800e..836d54a 100644
--- a/t/response/TestAPI/access2.pm
+++ b/t/response/TestAPI/access2.pm
@@ -101,9 +101,10 @@ PerlAddAuthzProvider my-group TestAPI::access2->authz_handler
     PerlResponseHandler Apache::TestHandler::ok1
     SetHandler modperl
 
+	<RequireAll>
     <IfModule @ACCESS_MODULE@>
         # needed to test $r->satisfies
-        Allow from All
+        Require all granted
     </IfModule>
     AuthType Basic
     AuthName "Access"
@@ -112,7 +113,7 @@ PerlAddAuthzProvider my-group TestAPI::access2->authz_handler
     <Limit POST>
        Require valid-user
     </Limit>
-    Satisfy All
+	</RequireAll>
     <IfModule @AUTH_MODULE@>
         # htpasswd -mbc auth-users goo foo
         # htpasswd -mb auth-users bar mar
diff --git a/t/response/TestModules/proxy.pm b/t/response/TestModules/proxy.pm
index ab59026..91e251e 100644
--- a/t/response/TestModules/proxy.pm
+++ b/t/response/TestModules/proxy.pm
@@ -47,9 +47,7 @@ __END__
     <IfModule mod_proxy.c>
         <Proxy http://@servername@:@port@/*>
             <IfModule @ACCESS_MODULE@>
-                Order Deny,Allow
-                Deny from all
-                Allow from @servername@
+                Require host @servername@
             </IfModule>
         </Proxy>
 
