--- bug33414-1.phpt.ori	2016-08-16 15:09:36.458850237 +0800
+++ php-5.2.14/ext/date/tests/bug33414-1.phpt	2016-08-16 15:15:21.980930748 +0800
@@ -1,10 +1,11 @@
 --TEST--
 Bug #33414 [1] (Comprehensive list of incorrect days returned after strotime() / date() tests)
+--INI--
+date.timezone=America/Mendoza
 --FILE--
 <?php 
 
 print "TZ=America/Mendoza - wrong day.\n";
-putenv("TZ=America/Mendoza");
 $tStamp = mktime (17, 17, 17, 1, 8327, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Sunday", $tStamp);
@@ -12,7 +13,7 @@
 print "wanted=Sunday              00:00:00\n\n";
 
 print "TZ=America/Catamarca - wrong day.\n";
-putenv("TZ=America/Catamarca");
+date_default_timezone_set("America/Catamarca");
 $tStamp = mktime (17, 17, 17, 1, 7599, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Sunday", $tStamp);
@@ -20,7 +21,7 @@
 print "wanted=Sunday              00:00:00\n\n";
 
 print "TZ=America/Cordoba - wrong day.\n";
-putenv("TZ=America/Cordoba");
+date_default_timezone_set("America/Cordoba");
 $tStamp = mktime (17, 17, 17, 1, 7599, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Sunday", $tStamp);
@@ -28,7 +29,7 @@
 print "wanted=Sunday              00:00:00\n\n";
 
 print "TZ=America/Rosario - wrong day.\n";
-putenv("TZ=America/Rosario");
+date_default_timezone_set("America/Rosario");
 $tStamp = mktime (17, 17, 17, 1, 7958, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Tuesday", $tStamp);
@@ -37,7 +38,7 @@
 
 print "TZ=Europe/Vienna - wrong day - giving unexpected results, at
 least on my system :-)\n";
-putenv("TZ=Europe/Vienna");
+date_default_timezone_set("Europe/Vienna");
 $tStamp = mktime (17, 17, 17, 1, 3746, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Thursday", $tStamp);
@@ -45,15 +46,15 @@
 print "wanted=Thursday             00:00:00\n\n";
 
 print "TZ=Asia/Baku - wrong day.\n";
-putenv("TZ=Asia/Baku");
+date_default_timezone_set("Asia/Baku");
 $tStamp = mktime (17, 17, 17, 1, 8299, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
-$strtotime_tstamp = strtotime("next Sunday", $tStamp);
+$strtotime_tstamp = strtotime("second Monday", $tStamp);
 print "result=".date("l Y-m-d H:i:s T I", $strtotime_tstamp)."\n";
-print "wanted=Sunday              00:00:00\n\n";
+print "wanted=Monday              00:00:00\n\n";
 
 print "TZ=America/Noronha - wrong day.\n";
-putenv("TZ=America/Noronha");
+date_default_timezone_set("America/Noronha");
 $tStamp = mktime (17, 17, 17, 1, 10866, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Friday", $tStamp);
@@ -61,7 +62,7 @@
 print "wanted=Friday              00:00:00\n\n";
 
 print "TZ=America/Havana - wrong day.\n";
-putenv("TZ=America/Havana");
+date_default_timezone_set("America/Havana");
 $tStamp = mktime (17, 17, 17, 1, 12720, 1970);  
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Thursday", $tStamp);
@@ -69,7 +70,7 @@
 print "wanted=Thursday             00:00:00\n\n";
 
 print "TZ=Europe/Tallinn - wrong day.\n";
-putenv("TZ=Europe/Tallinn");   
+date_default_timezone_set("Europe/Tallinn");   
 $tStamp = mktime (17, 17, 17, 1, 11777, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Saturday", $tStamp);
@@ -77,7 +78,7 @@
 print "wanted=Saturday          00:00:00\n\n";  
 
 print "TZ=Asia/Jerusalem - wrong day.\n";     
-putenv("TZ=Asia/Jerusalem");
+date_default_timezone_set("Asia/Jerusalem");
 $tStamp = mktime (17, 17, 17, 1, 13056, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Thursday", $tStamp);
@@ -85,7 +86,7 @@
 print "wanted=Thursday             00:00:00\n\n";         
 
 print "TZ=Europe/Vilnius - wrong day.\n";
-putenv("TZ=Europe/Vilnius");
+date_default_timezone_set("Europe/Vilnius");
 $tStamp = mktime (17, 17, 17, 1, 12140, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Friday", $tStamp);
@@ -93,7 +94,7 @@
 print "wanted=Friday            00:00:00\n\n";
 
 print "TZ=Pacific/Kwajalein - wrong day.\n";
-putenv("TZ=Pacific/Kwajalein");
+date_default_timezone_set("Pacific/Kwajalein");
 $tStamp = mktime (17, 17, 17, 1, 8626, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Friday", $tStamp);
@@ -101,7 +102,7 @@
 print "wanted=Friday            00:00:00\n\n";
 
 print "TZ=Asia/Ulan_Bator - wrong day.\n";
-putenv("TZ=Asia/Ulan_Bator");
+date_default_timezone_set("Asia/Ulan_Bator");
 $tStamp = mktime (17, 17, 17, 1, 11588, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Saturday", $tStamp);
@@ -109,7 +110,7 @@
 print "wanted=Saturday            00:00:00\n\n";
 
 print "TZ=America/Cancun - wrong day.\n";
-putenv("TZ=America/Cancun");
+date_default_timezone_set("America/Cancun");
 $tStamp = mktime (17, 17, 17, 1, 11785, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Sunday", $tStamp);
@@ -117,7 +118,7 @@
 print "wanted=Sunday            00:00:00\n\n";
 
 print "TZ=America/Mexico_City - wrong day.\n";
-putenv("TZ=America/Mexico_City");
+date_default_timezone_set("America/Mexico_City");
 $tStamp = mktime (17, 17, 17, 1, 11781, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Wednesday", $tStamp);
@@ -125,7 +126,7 @@
 print "wanted=Wednesday            00:00:00\n\n";
 
 print "TZ=America/Mazatlan - wrong day.\n";
-putenv("TZ=America/Mazatlan");
+date_default_timezone_set("America/Mazatlan");
 $tStamp = mktime (17, 17, 17, 1, 11780, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Tuesday", $tStamp);
@@ -133,7 +134,7 @@
 print "wanted=Tuesday            00:00:00\n\n";
 
 print "TZ=America/Chihuahua - wrong day.\n";
-putenv("TZ=America/Chihuahua");
+date_default_timezone_set("America/Chihuahua");
 $tStamp = mktime (17, 17, 17, 1, 11782, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Thursday", $tStamp);
@@ -141,7 +142,7 @@
 print "wanted=Thursday            00:00:00\n\n";
 
 print "TZ=Asia/Kuala_Lumpur - wrong day.\n";     
-putenv("TZ=Asia/Kuala_Lumpur");
+date_default_timezone_set("Asia/Kuala_Lumpur");
 $tStamp = mktime (17, 17, 17, 1, 4380, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Monday", $tStamp);
@@ -149,7 +150,7 @@
 print "wanted=Monday            00:00:00\n\n";            
 
 print "TZ=Pacific/Chatham - wrong day.\n";       
-putenv("TZ=Pacific/Chatham");  
+date_default_timezone_set("Pacific/Chatham");  
 $tStamp = mktime (17, 17, 17, 1, 1762, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Monday", $tStamp);
@@ -157,7 +158,7 @@
 print "wanted=Monday            00:00:00\n\n";            
 
 print "TZ=America/Lima - wrong day.\n";        
-putenv("TZ=America/Lima");   
+date_default_timezone_set("America/Lima");   
 $tStamp = mktime (17, 17, 17, 1, 5839, 1970); 
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Thursday", $tStamp);
@@ -165,7 +166,7 @@
 print "wanted=Thursday            00:00:00\n\n";          
 
 print "TZ=Asia/Karachi - wrong day.\n";
-putenv("TZ=Asia/Karachi");
+date_default_timezone_set("Asia/Karachi");
 $tStamp = mktime (17, 17, 17, 1, 11783, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Friday", $tStamp);
@@ -173,7 +174,7 @@
 print "wanted=Friday            00:00:00\n\n";
 
 print "TZ=America/Asuncion - wrong day.\n";
-putenv("TZ=America/Asuncion");
+date_default_timezone_set("America/Asuncion");
 $tStamp = mktime (17, 17, 17, 1, 11746, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Wednesday", $tStamp);
@@ -181,7 +182,7 @@
 print "wanted=Wednesday            00:00:00\n\n";
 
 print "TZ=Asia/Singapore - wrong day.\n";
-putenv("TZ=Asia/Singapore");
+date_default_timezone_set("Asia/Singapore");
 $tStamp = mktime (17, 17, 17, 1, 4383, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Thursday", $tStamp);
@@ -189,7 +190,7 @@
 print "wanted=Thursday             00:00:00\n\n";
 
 print "TZ=America/Montevideo - wrong day.\n";
-putenv("TZ=America/Montevideo");
+date_default_timezone_set("America/Montevideo");
 $tStamp = mktime (17, 17, 17, 1, 12678, 1970);
 print "tStamp=". date("l Y-m-d H:i:s T I", $tStamp). "\n";
 $strtotime_tstamp = strtotime("next Thursday", $tStamp);
@@ -226,8 +227,8 @@
 
 TZ=Asia/Baku - wrong day.
 tStamp=Sunday 1992-09-20 17:17:17 AZST 1
-result=Sunday 1992-09-27 00:00:00 AZT 0
-wanted=Sunday              00:00:00
+result=Monday 1992-09-28 00:00:00 AZT 0
+wanted=Monday              00:00:00
 
 TZ=America/Noronha - wrong day.
 tStamp=Friday 1999-10-01 17:17:17 FNT 0
