--- libhugetlbfs-2.17/Makefile.old	2015-02-15 17:15:08.140579203 +0800
+++ libhugetlbfs-2.17/Makefile	2015-02-15 17:16:19.146583819 +0800
@@ -104,10 +104,27 @@
 TMPLIB64 = lib64
 TMPLIB32 = lib
 else
+ifeq ($(ARCH),ppc64)
+CC64 = $(CC) -m64
+ELF64 = elf64ppc
+TMPLIB64 = lib64
+TMPLIB32 = lib
+else
+ifeq ($(ARCH),s390)
+CC64 = $(CC) -m64
+CC32 = $(CC) -m31
+ELF32 = elf_s390
+ELF64 = elf64_s390
+TMPLIB64 = lib64
+TMPLIB32 = lib
+CUSTOM_LDSCRIPTS = no
+else
 $(error "Unrecognized architecture ($(ARCH))")
 endif
 endif
 endif
+endif
+endif
 endif
 endif
 endif
