--- qa_nfstests-04/server.orig	2012-11-29 10:59:24.000000000 +0100
+++ qa_nfstests-04/server	2012-11-29 11:05:29.000000000 +0100
@@ -18,7 +18,8 @@
 #
 Program=`basename $0`
 
-InitFile="./tests.init"
+PTH="/usr/share/qa/qa_nfstests/bin"
+InitFile="$PTH/tests.init"
 USAGE="usage:  $Program [-a|-b|-g|-s|-l|-c] [-f|-t|-n|-h] [-o mnt_options] [-p server_path] [-m mntpoint] [-N passes] server_name"
 
 # defaults
