diff -Nur qa_test_bind.org/qa_test_bind/data/secondary/named.conf qa_test_bind/qa_test_bind/data/secondary/named.conf
--- qa_test_bind.org/qa_test_bind/data/secondary/named.conf	2012-03-14 14:27:49.043542791 +0800
+++ qa_test_bind/data/qa_test_bind/secondary/named.conf	2012-08-22 16:45:39.636364806 +0800
@@ -84,10 +84,10 @@
 	file "root.hint";
 };
 
-#zone "suse.cz" {
-#        type slave;
-#        file "slave/suse.cz";
-#        masters { 10.20.0.2; };
-#
-#};
+zone "suse.cz" {
+        type slave;
+        file "slave/suse.cz";
+        masters { 10.20.0.2; };
+
+};
 
diff -Nur qa_test_bind.org/qa_test_bind/data/secondary/slave/suse.cz qa_test_bind/qa_test_bind/data/secondary/slave/suse.cz
--- qa_test_bind.org/qa_test_bind/data/secondary/slave/suse.cz	1970-01-01 08:00:00.000000000 +0800
+++ qa_test_bind/qa_test_bind/data/secondary/slave/suse.cz	2012-08-22 16:47:08.104489578 +0800
@@ -0,0 +1,9 @@
+$ttl 38400
+suse.cz.		IN	SOA	tested.suse.cz. testing.suse.cz. (
+			994077097
+			10800
+			3600
+			432000
+			38400 )
+suse.cz.		IN	NS	tested.suse.cz.
+felix.suse.cz.		IN	A	10.20.1.89
