diff -Nurp kernbench-0.50.org/kernbench kernbench-0.50/kernbench
--- kernbench-0.50.org/kernbench	2009-12-11 22:42:08.000000000 +0800
+++ kernbench-0.50/kernbench	2015-07-21 18:33:50.001682542 +0800
@@ -241,7 +241,7 @@ tempjobs=$custjobs
 runname="Custom load -j $custjobs"
 do_runs
 
-temp_runs=$max_runs
-tempjobs=""
-runname="Maximal load -j"
-do_runs
+#temp_runs=$max_runs
+#tempjobs=""
+#runname="Maximal load -j"
+#do_runs
