diff --git a/sysbench-example-tests/sysbench-example-all-tests b/sysbench-example-tests/sysbench-example-all-tests
index 98d26ce..5485e07 100755
--- a/sysbench-example-tests/sysbench-example-all-tests
+++ b/sysbench-example-tests/sysbench-example-all-tests
@@ -2,10 +2,10 @@
 # this script is a placeholder to run specific sysbench variations
 
 # this one is for very high load, kernel should only survive :-)
-for i in 1 2 3 4 5 6 7 8 9; do sysbench --max-time=86400 --num-threads=409 --max-requests=1000000 --thread-stack-size=512 --init-rng=on --validate=on   --test=fileio --file-total-size=1024M --file-test-mode=rndwr run & done
+for i in 1 2 3 4 5 6 7 8 9; do sysbench --max-time=86400 --num-threads=200 --max-requests=1000000 --thread-stack-size=512 --init-rng=on --validate=on   --test=fileio --file-total-size=1024M --file-test-mode=rndwr run & done
 
 # this one trys to coverage all available sysbench parameters
-for j in 1 2 4 8 16 32 64 128 256; do
+for j in 1 2 4 8 16 32 64 128 200; do
 	for i in cpu threads fileio memory mutex oltp; do sysbench-example-test -t $i -T $j; done
 done
 
