--- sysbench/tests/fileio/sb_fileio.c
+++ sysbench/tests/fileio/sb_fileio.c
@@ -1588,6 +1588,8 @@
 #ifdef HAVE_POSIX_MEMALIGN
   page_size = sb_getpagesize();
   posix_memalign((void **)&buffer, page_size, size);
+  if (posix_memalign((void **)&buffer, page_size, file_max_request_size))
+	  exit(255);
 #elif defined(HAVE_MEMALIGN)
   page_size = sb_getpagesize();
   buffer = memalign(page_size, size);
