--- distro-scripts/install-opensuse-packages.sh	2009-11-16 01:13:39.000000000 +0800
+++ phoronix-test-suite/pts/distro-scripts/install-opensuse-packages.sh	2011-03-03 15:53:52.445969095 +0800
@@ -2,12 +2,12 @@
 
 # OpenSuSE package installation
 
-echo "Please enter your root password below:" 1>&2
+#echo "Please enter your root password below:" 1>&2
 
 if [ -x /usr/bin/zypper ]; then
-	su root -c "zypper install -l --force-resolution $*"
+	zypper install -l --force-resolution -y $*
 else
-	su root -c "yast -i $*"
+	yast -i $*
 fi
 
 exit
