--- qa_test_openssl-1.0.1g.orig/crypto/ecdh/ecdhtest.c	2014-03-18 00:14:20.000000000 +0800
+++ qa_test_openssl-1.0.1g/crypto/ecdh/ecdhtest.c	2015-11-20 16:24:54.951280330 +0800
@@ -323,11 +323,15 @@
 	if ((ctx=BN_CTX_new()) == NULL) goto err;
 
 	/* NIST PRIME CURVES TESTS */
+#if 0
 	if (!test_ecdh_curve(NID_X9_62_prime192v1, "NIST Prime-Curve P-192", ctx, out)) goto err;
 	if (!test_ecdh_curve(NID_secp224r1, "NIST Prime-Curve P-224", ctx, out)) goto err;
+#endif
 	if (!test_ecdh_curve(NID_X9_62_prime256v1, "NIST Prime-Curve P-256", ctx, out)) goto err;
 	if (!test_ecdh_curve(NID_secp384r1, "NIST Prime-Curve P-384", ctx, out)) goto err;
+#if 0
 	if (!test_ecdh_curve(NID_secp521r1, "NIST Prime-Curve P-521", ctx, out)) goto err;
+#endif
 #ifndef OPENSSL_NO_EC2M
 	/* NIST BINARY CURVES TESTS */
 	if (!test_ecdh_curve(NID_sect163k1, "NIST Binary-Curve K-163", ctx, out)) goto err;
