Index: qa_test_openssl-1.0.1q/crypto/ecdh/ecdhtest.c
===================================================================
--- qa_test_openssl-1.0.1q.orig/crypto/ecdh/ecdhtest.c
+++ qa_test_openssl-1.0.1q/crypto/ecdh/ecdhtest.c
@@ -337,18 +337,22 @@ int main(int argc, char *argv[])
         goto err;
 
     /* NIST PRIME CURVES TESTS */
+#if 0
     if (!test_ecdh_curve
         (NID_X9_62_prime192v1, "NIST Prime-Curve P-192", ctx, out))
         goto err;
     if (!test_ecdh_curve(NID_secp224r1, "NIST Prime-Curve P-224", ctx, out))
         goto err;
+#endif
     if (!test_ecdh_curve
         (NID_X9_62_prime256v1, "NIST Prime-Curve P-256", ctx, out))
         goto err;
     if (!test_ecdh_curve(NID_secp384r1, "NIST Prime-Curve P-384", ctx, out))
         goto err;
+#if 0
     if (!test_ecdh_curve(NID_secp521r1, "NIST Prime-Curve P-521", ctx, out))
         goto err;
+#endif
 # ifndef OPENSSL_NO_EC2M
     /* NIST BINARY CURVES TESTS */
     if (!test_ecdh_curve(NID_sect163k1, "NIST Binary-Curve K-163", ctx, out))
