Index: qa_test_openssl-1.0.1q/crypto/rc4/rc4test.c
===================================================================
--- qa_test_openssl-1.0.1q.orig/crypto/rc4/rc4test.c
+++ qa_test_openssl-1.0.1q/crypto/rc4/rc4test.c
@@ -121,12 +121,6 @@ int main(int argc, char *argv[])
     RC4_KEY key;
     unsigned char obuf[512];
 
-# if !defined(OPENSSL_PIC)
-    void OPENSSL_cpuid_setup(void);
-
-    OPENSSL_cpuid_setup();
-# endif
-
     for (i = 0; i < 6; i++) {
         RC4_set_key(&key, keys[i][0], &(keys[i][1]));
         memset(obuf, 0x00, sizeof(obuf));
