diff -Naur tests.old/regression/subdomain/prologue.inc tests/regression/subdomain/prologue.inc
--- tests.old/regression/subdomain/prologue.inc	2007-12-23 02:10:29.000000000 +0100
+++ tests/regression/subdomain/prologue.inc	2008-11-10 13:09:46.000000000 +0100
@@ -146,7 +146,7 @@
 	# Suck. SuSE's ldd has a line "linux-gate.so.1 =>  (0xffffe000)"
 	#       Red Hat's ldd has "/lib/ld-linux.so.2 (0x007b1000)"
 	# good ${DIETY}, what gross kludgage.
-	libs=$(ldd $exec | egrep -v "linux-(vdso(32|64)|gate).so.1" | sed 's~^.*=> \(/.*\) (.*$~\1~' | awk '{print $1}')
+	libs=$(ldd $exec | egrep -v "linux-(vdso(32|64|)|gate).so.1" | sed 's~^.*=> \(/.*\) (.*$~\1~' | awk '{print $1}')
 
 	dynlibs="/etc/ld.so.cache:r"
 
