Index: tests/regression/subdomain/clone.c
===================================================================
--- tests.orig/regression/subdomain/clone.c
+++ tests/regression/subdomain/clone.c
@@ -61,7 +61,7 @@ int main(int argc, char *argv[])
 	int waitstatus;
 	int c;
 	char buf[BUFSIZ];
-	int stack_size = PAGE_SIZE << 4;
+	int stack_size = getpagesize()<< 4;
 	void *child_stack = malloc(stack_size);
 	int clone_flags = SIGCHLD;
 
