From e1f32ce1ff8f7038806c9565d6dd9c1c8644b51e Mon Sep 17 00:00:00 2001
From: Cyril Hrubis <chrubis@suse.cz>
Date: Wed, 11 Sep 2013 11:48:01 +0200
Subject: [PATCH] runtest/mm: Fix ksn -> ksm typo.

Signed-off-by: Cyril Hrubis <chrubis@suse.cz>
Acked-by: Wanlong Gao <gaowanlong@cn.fujitsu.com>
---
 runtest/mm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/runtest/mm b/runtest/mm
index a39becd..28151b5 100644
--- a/runtest/mm
+++ b/runtest/mm
@@ -72,7 +72,7 @@ ksm04_1 ksm04 -u 128
 ksm05 ksm05 -I 10
 ksm06 ksm06
 ksm06_1 ksm06 -n 10
-ksn06_2 ksm06 -n 10000
+ksm06_2 ksm06 -n 10000
 
 cpuset01 cpuset01 -I 3600
 
-- 
1.8.1.5

