CC: Frank Seidel <fseidel@suse.de>
Date: Thu Jan 19 13:32:59 CET 2006
Subject: Enable fsstress test in runttest/fs

TODO: Review for upstream merge? fsstress internal only?


Author: Frank Seidel <fseidel@suse.de>
Signed-off-by: Frank Seidel <fseidel@suse.de>

---

Index: ltp-full-20080825/runtest/fs
===================================================================
--- ltp-full-20080825.orig/runtest/fs
+++ ltp-full-20080825/runtest/fs
@@ -57,6 +57,7 @@ ftest05 ftest05
 ftest06 ftest06
 ftest07 ftest07
 ftest08 ftest08
+fsstress01 fsstress -c -d /tmp/ -l 100 -p 1000
 
 lftest01	lftest 100
 writetest01	writetest
