Index: ltp-full-20100831/runtest/syscalls
===================================================================
--- ltp-full-20100831.orig/runtest/syscalls
+++ ltp-full-20100831/runtest/syscalls
@@ -562,10 +562,10 @@ modify_ldt02 modify_ldt02
 # to run correctly. Please see individual test
 # code for more information.
 #
-mount01 mount01 -D DEVICE -T DEVICE_FS_TYPE
-mount02 mount02 -D DEVICE -T DEVICE_FS_TYPE
-mount03 mount03 -D DEVICE -T DEVICE_FS_TYPE
-mount04 mount04 -D DEVICE -T DEVICE_FS_TYPE
+mount01 _mount mount01 
+mount02 _mount mount02 
+mount03 _mount mount03 
+mount04 _mount mount04 
 
 mount1234 test_mount
 
Index: ltp-full-20100831/testcases/kernel/syscalls/mount/_mount
===================================================================
--- /dev/null
+++ ltp-full-20100831/testcases/kernel/syscalls/mount/_mount
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+if [ $# -ne 1 ]; then
+        echo "please enter a testcase name as input: mount(01|02|03|04)"
+        exit 1
+fi
+
+case $1 in
+        mount0[1-4])
+                        ;;
+        *)
+                echo "invalid input, please input mount0[1-4]"
+                exit 1
+                ;;
+esac
+
+
+dd if=/dev/zero of=./test_loop_device bs=1M count=512
+
+if [ $? -ne 0 ]; then
+        echo 'Run dd if=/dev/zero of=./test_loop_device bs=1M count=512 faile, please check space'
+        exit 1
+fi
+
+loop_device_file=`losetup -f`
+losetup $loop_device_file ./test_loop_device
+mkfs.ext3 $loop_device_file
+
+if [ $? -ne 0 ]; then
+        echo "Run mkfs.ext3 $loop_device_file failed"
+        exit 1
+fi
+
+$1 -D $loop_device_file -T ext3
+
+losetup -d $loop_device_file
+
+rm ./test_loop_device
+
