--- ltp-full-20081031/testcases/kernel/sched/sched_stress/sched_stress.sh.orig	2008-12-10 16:09:42.000000000 +0100
+++ ltp-full-20081031/testcases/kernel/sched/sched_stress/sched_stress.sh	2008-12-10 16:09:49.000000000 +0100
@@ -30,6 +30,6 @@ export KERNEL=./sched_datafile
 touch $KERNEL
 echo 0.000000 > sch.measure
 export RAWDEV=`df / | grep dev | awk {'print $1'}`
-sched_driver -s 0.9 -t 0.02 -p 2 > /tmp/tmp$$
+./sched_driver -s 0.9 -p 2 > /tmp/tmp$$
 tail -n 5 /tmp/tmp$$
 rm -rf /tmp/tmp$$ ./sched_datafile sch.measure
