Index: ltp-full-20090930/ctcstools/ltp_wrapper.sh
===================================================================
--- ltp-full-20090930.orig/ctcstools/ltp_wrapper.sh
+++ ltp-full-20090930/ctcstools/ltp_wrapper.sh
@@ -59,30 +59,30 @@ elif [ `basename $0` == "ltp_wrapper.sh"
 fi;
 
 #eval $@
-eval pan -n $$ -a $$ -e -S -L -f $@ 
+eval ltp-pan -n $$ -a $$ -e -S -L -f $@
 rc=$?
  
 if [ $rc -eq 0 ]; then
-  echo "INFO: pan reported all tests PASS"
+  echo "INFO: ltp-pan reported all tests PASS"
   VALUE=0
 # LTP_MODE map unsuitable/unsupported tests to PASS. Check if _only_ TCONF is set!
 elif [ $LTP_MODE -eq 1 ] && [[ "$rc" -eq 4 ]]; then
-  echo "INFO: pan reported this test is unsuitable/unsupported (LTP_MODE)"
+  echo "INFO: ltp-pan reported this test is unsuitable/unsupported (LTP_MODE)"
   VALUE=0
 # LTP_MODE uses bit fields - TFAIL is 0x1
 elif [ $LTP_MODE -eq 1 ]  && [ "$(( $rc & 1 ))" -eq 0 ] ; then
-  echo "INFO: pan reported this test had INTERNAL ERRORS (LTP_MODE)"
+  echo "INFO: ltp-pan reported this test had INTERNAL ERRORS (LTP_MODE)"
   VALUE=2
 # OPENPOSIX_MODE map internal errors
 elif [ $OPENPOSIX_MODE -eq 1 ] && [[ "$rc" -eq 5 || "$rc" -eq "2" ]]; then
-  echo "INFO: pan reported this test had INTERNAL ERRORS (OPENPOSIX_MODE)"
+  echo "INFO: ltp-pan reported this test had INTERNAL ERRORS (OPENPOSIX_MODE)"
   VALUE=2
 # OPENPOSIX_MODE map unsuitable/unsupported tests to PASS
 elif [ $OPENPOSIX_MODE -eq 1 ] && [[ "$rc" -eq 4 ]]; then
-  echo "INFO: pan reported this test is unsuitable/unsupported (OPENPOSIX_MODE)"
+  echo "INFO: ltp-pan reported this test is unsuitable/unsupported (OPENPOSIX_MODE)"
   VALUE=0
 else
-  echo "INFO: pan reported some tests FAIL"
+  echo "INFO: ltp-pan reported some tests FAIL"
   VALUE=1
 fi
 
