Index: ltp-full-git/testcases/kernel/syscalls/Makefile
===================================================================
--- ltp-full-git.orig/testcases/kernel/syscalls/Makefile
+++ ltp-full-git/testcases/kernel/syscalls/Makefile
@@ -29,4 +29,6 @@ FILTER_OUT_DIRS	+= capget capset chmod c
 		   nftw profil remap_file_pages
 endif
 
+FILTER_OUT_DIRS += add_key
+
 include $(top_srcdir)/include/mk/generic_trunk_target.mk
