--- server-tests-config.php	2012-04-05 11:08:23.926944163 +0800
+++ php-5.3.8/server-tests-config.php	2012-04-05 11:08:30.742944450 +0800
@@ -9,7 +9,7 @@
    
 $conf = array(
 /* path to the php source tree */
-'TEST_PHP_SRCDIR'      =>    NULL,
+'TEST_PHP_SRCDIR'      =>    '/usr/share/qa/qa_test_php53',
 
 /* executable that will be tested.  Not used for
    web based tests */
@@ -17,24 +17,25 @@
 
 /* php.ini to use when executing php */
 'PHPRC'                =>    NULL,
+'PHPRC'                =>    '/etc/php.ini',
 
 /* log format */
 'TEST_PHP_LOG_FORMAT'  =>    'LEODC',
 
 /* debugging detail in output. */
-'TEST_PHP_DETAILED'    =>    0,
+'TEST_PHP_DETAILED'    =>    1,
 
 /* error style for editors or IDE's */
 'TEST_PHP_ERROR_STYLE' =>    'EMACS',
 
-'REPORT_EXIT_STATUS'   =>    0,
+'REPORT_EXIT_STATUS'   =>    1,
 'NO_PHPTEST_SUMMARY'   =>    0,
 
 /* don't ask, and don't send results to QA if true */
 'NO_INTERACTION'       =>    true,
 
 /* base url prefixed to any requests */
-'TEST_WEB_BASE_URL'    =>    NULL,
+'TEST_WEB_BASE_URL'    =>    'http://localhost/php53-tests',
 
 /* if set, copy phpt files into this directory,
    which should be accessable via an http server.  The
@@ -48,7 +49,7 @@
    TEST_WEB_BASE_URL=http://localhost/test
    TEST_BASE_PATH=/path/to/htdocs/test
 */
-'TEST_BASE_PATH'   =>    NULL,
+'TEST_BASE_PATH'   =>    '/srv/www/htdocs/php53-tests',
 
 /* file extension of pages requested via http
    this allows for php to be configured to parse
@@ -60,6 +61,7 @@
 'TEST_CONTEXT_INFO'    =>    false,
 
 /* : or ; seperated list of paths */
+'TEST_WEB'             =>    true,
 'TEST_PATHS'           =>    NULL
 /* additional configuration items that may be set
    to provide proxy support for testes:
@@ -71,4 +73,4 @@
 */
 );
 
-?>
\ No newline at end of file
+?>
