diff -Naur memeat-0.1.old/memeat.c memeat-0.1/memeat.c
--- memeat-0.1.old/memeat.c	2008-10-15 12:10:40.000000000 +0200
+++ memeat-0.1/memeat.c	2008-10-15 12:11:56.000000000 +0200
@@ -9,13 +9,19 @@
 
 	size_t linelen = 0;
 	char *line = NULL;
-	unsigned long freemem = 0,i,w,count;
+	unsigned long freemem = 0, lowmem = 0, i,w,count;
 	FILE *f = fopen("/proc/meminfo", "r");
 	while (getdelim(&line, &linelen, '\n', f) > 0) { 
-		if (sscanf(line, "LowFree: %lu", &freemem) == 1)
-			break; 
+		sscanf(line, "LowFree: %lu", &lowmem);
+		sscanf(line, "MemFree: %lu", &freemem);
 	} 
 
+	/* If system is configured with HIGH memory, use LowFree,
+	* else use MemFree */
+	if (lowmem) {
+		freemem = lowmem;
+	}
+
 	freemem *= 1024; 
 	freemem -= freemem/20;
 
