diff -uNpr qa_test_nfs/nfs_start_stop.sh qa_nfs_/nfs_start_stop.sh
--- qa_test_nfs/nfs_start_stop.sh	2008-06-03 16:56:44.000000000 +0800
+++ qa_nfs_/nfs_start_stop.sh	2009-12-17 16:05:40.000000000 +0800
@@ -24,8 +24,8 @@ function testStop() {
 	return 0
     fi
 
-    if [ -n "`/etc/init.d/$SERVICE status | grep -e mountd -e statd -e nfsd | grep running`" ]; then
-	printMessage $MSG_FAILED "Still running (mountd, statd, or nfsd)"
+    if [ -n "`/etc/init.d/$SERVICE status | grep -e mountd  -e nfsd | grep running`" ]; then
+	printMessage $MSG_FAILED "Still running (mountd or nfsd)"
 	return 1
     fi
 
diff -uNpr qa_test_nfs/nfs_usermapping_rootsquash.sh qa_nfs_/nfs_usermapping_rootsquash.sh
--- qa_test_nfs/nfs_usermapping_rootsquash.sh	2008-06-05 20:36:26.000000000 +0800
+++ qa_nfs_/nfs_usermapping_rootsquash.sh	2009-12-17 16:36:26.000000000 +0800
@@ -72,7 +72,7 @@ function init() {
         printMessage $MSG_ERROR "Backup and rewrite config /etc/exports by $EXPORTSFILE1"
         exit $FAILED
     fi
-    if ! addUser "$TESTUSER" ; then
+    if ! addUser "$TESTUSER" "$TESTUSERGROUP" ; then
         printMessage $MSG_ERROR "Make a new user"
         exit $FAILED
     fi

diff -uNpr qa_test_nfs/nfs_start_stop.sh qa_nfs_/nfs_start_stop.sh
--- qa_test_nfs/nfs_start_stop.sh	2008-06-03 16:56:44.000000000 +0800
+++ qa_nfs_/nfs_start_stop.sh	2015-12-03 13:17:35.838388091 +0800
@@ -47,8 +47,8 @@ function cleanup() {
 }
 
 init
-testStart;TESTRESULT=$(($? || $TESTRESULT))
 testStop; TESTRESULT=$(($? || $TESTRESULT))
+testStart;TESTRESULT=$(($? || $TESTRESULT))
 cleanup
 
 if [ $TESTRESULT -eq 0 ]; then
