diff -pur libmicro-0.4.0-bak/libmicro.c libmicro-0.4.0/libmicro.c
--- libmicro-0.4.0-bak/libmicro.c	2007-07-02 23:17:45.000000000 +0200
+++ libmicro-0.4.0/libmicro.c	2009-09-24 12:16:38.000000000 +0200
@@ -1255,6 +1255,7 @@ print_histo(barrier_t *b)
 	if (b->ba_batches > b->ba_datasize)
 		(void) printf("#       %12s %12d\n", "data dropped",
 		    b->ba_batches - b->ba_datasize);
+	free(histo);
 }
 
 static void
