--- lynis-1.3.0/include/tests_filesystems       2011-12-25 15:55:27.000000000 +0100
+++ lynis-1.3.0_suse/include/tests_filesystems  2013-01-10 11:56:13.279292980 +0100
@@ -314,7 +314,7 @@
 	    if [ ! "${FIND1}" = "" ]; then
 	        logtext "Result: found ${FIND1}"
 	    	logtext "Test: Checking default options on ${FIND1}"
-	        FIND2=`tune2fs -l ${FIND1} | grep "^Default mount options" | grep "acl"` 
+	        FIND2=`/sbin/tune2fs -l ${FIND1} | grep "^Default mount options" | grep "acl"` 
 		if [ ! "${FIND2}" = "" ]; then
 		    logtext "Result: found ACL option in default mount options"
 		    FOUND=1
