--- lynis-1.3.0/include/tests_homedirs	2011-12-25 15:55:58.000000000 +0100
+++ lynis-1.3.0_suse/include/tests_homedirs	2013-01-10 17:50:44.652450705 +0100
@@ -54,7 +54,7 @@
     Register --test-no HOME-9310 --weight L --network NO --description "Checking for suspicious shell history files"
     if [ ${SKIPTEST} -eq 0 ]; then
 	if [ ! "${HOMEDIRS}" = "" ]; then
-	    if [ ${OS} = "Solaris" ]; then
+	    if [ "${OS}" = "Solaris" ]; then
 	        # Solaris doesn't support -maxdepth
 	        FIND=`find ${HOMEDIRS} -name ".*history" -not -type f -print`
 	      else
