Index: mce-inject-git_25_11_2009/Makefile
===================================================================
--- mce-inject-git_25_11_2009.orig/Makefile
+++ mce-inject-git_25_11_2009/Makefile
@@ -1,7 +1,7 @@
 prefix := /usr
 manprefix := ${prefix}/share
 CFLAGS := -Os -g -Wall
-LDFLAGS += -lpthread
+LDFLAGS += -pthread
 
 OBJ := mce.tab.o lex.yy.o mce-inject.o util.o
 GENSRC := mce.tab.c mce.tab.h lex.yy.c
