diff -ru apparmor.orig/tests/regression/apparmor/exec.sh apparmor/tests/regression/apparmor/exec.sh
--- apparmor.orig/tests/regression/apparmor/exec.sh	2012-01-12 19:29:10.000000000 +0800
+++ apparmor/tests/regression/apparmor/exec.sh	2015-11-23 17:05:42.909202738 +0800
@@ -10,13 +10,13 @@
 #=DESCRIPTION Runs exec() through ux, ix & px functionality
 
 pwd=`dirname $0`
-pwd=`cd $pwd ; /bin/pwd`
+pwd=`cd $pwd ; /usr/bin/pwd`
 
 bin=$pwd
 
 . $bin/prologue.inc
 
-file=/bin/true
+file=/usr/bin/true
 ok_ix_perm=rix
 badperm=r
 ok_ux_perm=ux
diff -ru apparmor.orig/tests/regression/apparmor/ptrace.sh apparmor/tests/regression/apparmor/ptrace.sh
--- apparmor.orig/tests/regression/apparmor/ptrace.sh	2010-12-21 04:29:10.000000000 +0800
+++ apparmor/tests/regression/apparmor/ptrace.sh	2015-11-23 17:05:55.253202738 +0800
@@ -32,154 +32,154 @@
 # -h transition child to ptrace_helper before doing ptrace (used to test
 #  x transitions with ptrace)
 # test base line of unconfined tracing unconfined
-runchecktest "test 1" pass -n 100 /bin/true
-runchecktest "test 1 -c" pass -c -n 100 /bin/true
+runchecktest "test 1" pass -n 100 /usr/bin/true
+runchecktest "test 1 -c" pass -c -n 100 /usr/bin/true
 runchecktest "test 1 -h" pass -h -n 100 $helper
 runchecktest "test 1 -hc" pass -h -c -n 100 $helper
-runchecktest "test 1 -h prog" pass -h -n 100 $helper /bin/true
-runchecktest "test 1 -hc prog" pass -h -c -n 100 $helper /bin/true
+runchecktest "test 1 -h prog" pass -h -n 100 $helper /usr/bin/true
+runchecktest "test 1 -hc prog" pass -h -c -n 100 $helper /usr/bin/true
 
 # test that unconfined can ptrace before profile attaches
-genprofile image=/bin/true
-runchecktest "test 2" pass -n 100 /bin/true
-runchecktest "test 2 -c" pass -c -n 100 /bin/true
+genprofile image=/usr/bin/true
+runchecktest "test 2" pass -n 100 /usr/bin/true
+runchecktest "test 2 -c" pass -c -n 100 /usr/bin/true
 runchecktest "test 2 -h" pass -h -n 100 $helper
 runchecktest "test 2 -hc" pass -h -c -n 100 $helper
-runchecktest "test 2 -h prog" pass -h -n 100 $helper /bin/true
-runchecktest "test 2 -hc prog" pass -h -c -n 100 $helper /bin/true
+runchecktest "test 2 -h prog" pass -h -n 100 $helper /usr/bin/true
+runchecktest "test 2 -hc prog" pass -h -c -n 100 $helper /usr/bin/true
 
 #unconfined tracing confined helper
 #confined helper asking unconfined process to ptrace it
 genprofile image=$helper
 runchecktest "test 3 -h" pass -h -n 100 $helper
 runchecktest "test 3 -hc " pass -h -c -n 100 $helper
-# can't exec /bin/true so fail
-runchecktest "test 3 -h prog" fail -h -n 100 $helper /bin/true
-runchecktest "test 3 -hc prog" fail -h -c -n 100 $helper /bin/true
+# can't exec /usr/bin/true so fail
+runchecktest "test 3 -h prog" fail -h -n 100 $helper /usr/bin/true
+runchecktest "test 3 -hc prog" fail -h -c -n 100 $helper /usr/bin/true
 
 # lack of 'r' perm is currently not working
 genprofile image=$helper $helper:ix
 runchecktest "test 4 -h" pass -h -n 100 $helper
 runchecktest "test 4 -hc " pass -h -c -n 100 $helper
-# can't exec /bin/true so fail
-runchecktest "test 4 -h prog" fail -h -n 100 $helper /bin/true
-runchecktest "test 4 -hc prog" fail -h -c -n 100 $helper /bin/true
+# can't exec /usr/bin/true so fail
+runchecktest "test 4 -h prog" fail -h -n 100 $helper /usr/bin/true
+runchecktest "test 4 -hc prog" fail -h -c -n 100 $helper /usr/bin/true
 
 genprofile image=$helper $helper:rix
 runchecktest "test 5 -h" pass -h -n 100 $helper
 runchecktest "test 5 -hc " pass -h -c -n 100 $helper
-# can't exec /bin/true so fail
-runchecktest "test 5 -h prog" fail -h -n 100 $helper /bin/true
-runchecktest "test 5 -hc prog" fail -h -c -n 100 $helper /bin/true
+# can't exec /usr/bin/true so fail
+runchecktest "test 5 -h prog" fail -h -n 100 $helper /usr/bin/true
+runchecktest "test 5 -hc prog" fail -h -c -n 100 $helper /usr/bin/true
 
-genprofile image=$helper $helper:ix /bin/true:rix
+genprofile image=$helper $helper:ix /usr/bin/true:rix
 runchecktest "test 6 -h" pass -h -n 100 $helper
 runchecktest "test 6 -hc " pass -h -c -n 100 $helper
-runchecktest "test 6 -h prog" pass -h -n 100 $helper /bin/true
-runchecktest "test 6 -hc prog" pass -h -c -n 100 $helper /bin/true
+runchecktest "test 6 -h prog" pass -h -n 100 $helper /usr/bin/true
+runchecktest "test 6 -hc prog" pass -h -c -n 100 $helper /usr/bin/true
 
 #traced child can ptrace_me to unconfined have unconfined trace them
-genprofile image=/bin/true
-runchecktest "test 7" pass -n 100 /bin/true
+genprofile image=/usr/bin/true
+runchecktest "test 7" pass -n 100 /usr/bin/true
 # pass - ptrace_attach is done in unconfined helper
-runchecktest "test 7 -c " pass -c -n 100 /bin/true
+runchecktest "test 7 -c " pass -c -n 100 /usr/bin/true
 runchecktest "test 7 -h" pass -h -n 100 $helper
 # pass - ptrace_attach is done in unconfined helper
 runchecktest "test 7 -hc " pass -h -c -n 100 $helper
-runchecktest "test 7 -h prog" pass -h -n 100 $helper /bin/true
-runchecktest "test 7 -hc prog" pass -h -c -n 100 $helper /bin/true
+runchecktest "test 7 -h prog" pass -h -n 100 $helper /usr/bin/true
+runchecktest "test 7 -hc prog" pass -h -c -n 100 $helper /usr/bin/true
 
-genprofile image=$helper $helper:ix /bin/true:rix
-runchecktest "test 7a" pass -n 100 /bin/true
+genprofile image=$helper $helper:ix /usr/bin/true:rix
+runchecktest "test 7a" pass -n 100 /usr/bin/true
 # pass - ptrace_attach is allowed from confined process to unconfined
-runchecktest "test 7a -c " pass -c -n 100 /bin/true
+runchecktest "test 7a -c " pass -c -n 100 /usr/bin/true
 runchecktest "test 7a -h" pass -h -n 100 $helper
 # pass - ptrace_attach is allowed from confined process to unconfined
 runchecktest "test 7a -hc " pass -h -c -n 100 $helper
-runchecktest "test 7a -h prog" pass -h -n 100 $helper /bin/true
-runchecktest "test 7a -hc prog" pass -h -c -n 100 $helper /bin/true
+runchecktest "test 7a -h prog" pass -h -n 100 $helper /usr/bin/true
+runchecktest "test 7a -hc prog" pass -h -c -n 100 $helper /usr/bin/true
 
 #traced helper from unconfined
-genprofile image=$helper $helper:ix /bin/true:rpx -- image=/bin/true
-runchecktest "test 8" pass -n 100 /bin/true
+genprofile image=$helper $helper:ix /usr/bin/true:rpx -- image=/usr/bin/true
+runchecktest "test 8" pass -n 100 /usr/bin/true
 # pass - ptrace_attach is done before exec
-runchecktest "test 8 -c " pass -c -n 100 /bin/true
+runchecktest "test 8 -c " pass -c -n 100 /usr/bin/true
 runchecktest "test 8 -h" pass -h -n 100 $helper
 runchecktest "test 8 -hc " pass -h -c -n 100 $helper
 # pass - can px if tracer can ptrace target
-runchecktest "test 8 -h prog" pass -h -n 100 $helper /bin/true
-runchecktest "test 8 -hc prog" pass -h -c -n 100 $helper /bin/true
+runchecktest "test 8 -h prog" pass -h -n 100 $helper /usr/bin/true
+runchecktest "test 8 -hc prog" pass -h -c -n 100 $helper /usr/bin/true
 
 #traced helper from unconfined
-genprofile image=$helper $helper:ix /bin/true:rux -- image=/bin/true
-runchecktest "test 9" pass -n 100 /bin/true
+genprofile image=$helper $helper:ix /usr/bin/true:rux -- image=/usr/bin/true
+runchecktest "test 9" pass -n 100 /usr/bin/true
 # pass - ptrace_attach is done before exec
-runchecktest "test 9 -c " pass -c -n 100 /bin/true
+runchecktest "test 9 -c " pass -c -n 100 /usr/bin/true
 runchecktest "test 9 -h" pass -h -n 100 $helper
 runchecktest "test 9 -hc " pass -h -c -n 100 $helper
 # pass - can ux if tracer can ptrace target
-runchecktest "test 9 -h prog" pass -h -n 100 $helper /bin/true
-runchecktest "test 9 -hc prog" pass -h -c -n 100 $helper /bin/true
+runchecktest "test 9 -h prog" pass -h -n 100 $helper /usr/bin/true
+runchecktest "test 9 -hc prog" pass -h -c -n 100 $helper /usr/bin/true
 
 genprofile
 # fail due to no exec permission
-runchecktest "test 10" fail -n 100 /bin/true
-runchecktest "test 10 -c" fail -c -n 100 /bin/true
+runchecktest "test 10" fail -n 100 /usr/bin/true
+runchecktest "test 10 -c" fail -c -n 100 /usr/bin/true
 runchecktest "test 10 -h" fail -h -n 100 $helper
 runchecktest "test 10 -hc" fail -h -c -n 100 $helper
-runchecktest "test 10 -h prog" fail -h -n 100 $helper /bin/true
-runchecktest "test 10 -hc prog" fail -h -c -n 100 $helper /bin/true
+runchecktest "test 10 -h prog" fail -h -n 100 $helper /usr/bin/true
+runchecktest "test 10 -hc prog" fail -h -c -n 100 $helper /usr/bin/true
 
-genprofile /bin/true:ix $helper:ix
+genprofile /usr/bin/true:ix $helper:ix
 # fail due to missing r permission
-#runchecktest "test 11" fail -n 100 /bin/true
-#runchecktest "test 11 -c" fail -c -n 100 /bin/true
+#runchecktest "test 11" fail -n 100 /usr/bin/true
+#runchecktest "test 11 -c" fail -c -n 100 /usr/bin/true
 #runchecktest "test 11 -h" fail -h -n 100 $helper
 #runchecktest "test 11 -hc" fail -h -c -n 100 $helper
-#runchecktest "test 11 -h prog" fail -h -n 100 $helper /bin/true
-#runchecktest "test 11 -hc prog" fail -h -c -n 100 $helper /bin/true
+#runchecktest "test 11 -h prog" fail -h -n 100 $helper /usr/bin/true
+#runchecktest "test 11 -hc prog" fail -h -c -n 100 $helper /usr/bin/true
 
 # pass allowed to ix self
-genprofile /bin/true:rix $helper:rix
-runchecktest "test 12" pass -n 100 /bin/true
-runchecktest "test 12 -c" pass -c -n 100 /bin/true
+genprofile /usr/bin/true:rix $helper:rix
+runchecktest "test 12" pass -n 100 /usr/bin/true
+runchecktest "test 12 -c" pass -c -n 100 /usr/bin/true
 runchecktest "test 12 -h" pass -h -n 100 $helper
 runchecktest "test 12 -hc" pass -h -c -n 100 $helper
-runchecktest "test 12 -h prog" pass -h -n 100 $helper /bin/true
-runchecktest "test 12 -hc prog" pass -h -c -n 100 $helper /bin/true
+runchecktest "test 12 -h prog" pass -h -n 100 $helper /usr/bin/true
+runchecktest "test 12 -hc prog" pass -h -c -n 100 $helper /usr/bin/true
 
 #ptraced confined app can't px - fails to unset profile
-genprofile image=$helper $helper:rix /bin/true:rpx
-runchecktest "test 13 -h prog" fail -h -n 100 $helper /bin/true
-runchecktest "test 13 -hc prog" fail -h -c -n 100 $helper /bin/true
+genprofile image=$helper $helper:rix /usr/bin/true:rpx
+runchecktest "test 13 -h prog" fail -h -n 100 $helper /usr/bin/true
+runchecktest "test 13 -hc prog" fail -h -c -n 100 $helper /usr/bin/true
 
 
 #ptraced confined app can ux - if the tracer is unconfined
 #
-genprofile image=$helper $helper:rix /bin/true:rux
-runchecktest "test 14a -h prog" pass -h -n 100 $helper /bin/true
-runchecktest "test 14a -hc prog" pass -h -c -n 100 $helper /bin/true
+genprofile image=$helper $helper:rix /usr/bin/true:rux
+runchecktest "test 14a -h prog" pass -h -n 100 $helper /usr/bin/true
+runchecktest "test 14a -hc prog" pass -h -c -n 100 $helper /usr/bin/true
 #ptraced confined app can't ux - if the tracer can't trace unconfined
-genprofile $helper:rpx -- image=$helper $helper:rix /bin/true:rux
-runchecktest "test 14b -h prog" fail -h -n 100 $helper /bin/true
-runchecktest "test 14b -hc prog" fail -h -c -n 100 $helper /bin/true
+genprofile $helper:rpx -- image=$helper $helper:rix /usr/bin/true:rux
+runchecktest "test 14b -h prog" fail -h -n 100 $helper /usr/bin/true
+runchecktest "test 14b -hc prog" fail -h -c -n 100 $helper /usr/bin/true
 
 #confined app can't ptrace an unconfined app
 genprofile $helper:rux
 runchecktest "test 15 -h" fail -h -n 100 $helper
-runchecktest "test 15 -h prog" fail -h -n 100 $helper /bin/true
+runchecktest "test 15 -h prog" fail -h -n 100 $helper /usr/bin/true
 #an unconfined app can't ask a confined app to trace it
 runchecktest "test 15 -hc" fail -h -c -n 100 $helper
-runchecktest "test 15 -hc prog" fail -h -c -n 100 $helper /bin/true
+runchecktest "test 15 -hc prog" fail -h -c -n 100 $helper /usr/bin/true
 
 #confined app can't ptrace an app confined by a different profile
 genprofile $helper:rpx -- image=$helper
 runchecktest "test 15 -h" fail -h -n 100 $helper
-runchecktest "test 15 -h prog" fail -h -n 100 $helper /bin/true
+runchecktest "test 15 -h prog" fail -h -n 100 $helper /usr/bin/true
 #a confined app can't ask another confined app with a different profile to
 #trace it
 runchecktest "test 15 -hc" fail -h -c -n 100 $helper
-runchecktest "test 15 -hc prog" fail -h -c -n 100 $helper /bin/true
+runchecktest "test 15 -hc prog" fail -h -c -n 100 $helper /usr/bin/true
 
 
 
diff -ru apparmor.orig/tests/regression/apparmor/regex.sh apparmor/tests/regression/apparmor/regex.sh
--- apparmor.orig/tests/regression/apparmor/regex.sh	2010-12-21 04:29:10.000000000 +0800
+++ apparmor/tests/regression/apparmor/regex.sh	2015-11-23 17:08:51.701202738 +0800
@@ -18,7 +18,7 @@
 # pcre stuff works as well.
 
 pwd=`dirname $0`
-pwd=`cd $pwd ; /bin/pwd`
+pwd=`cd $pwd ; /usr/bin/pwd`
 
 bin=$pwd
 
@@ -119,54 +119,54 @@
 runchecktest "GLOB FOLLOWED BY DOUBLE TAILGLOB (fail)" fail ${file2}
 
 settest exec
-file=/bin/true
+file=/usr/bin/true
 okperm=rix
 baderm=r
 
 # PASS TEST - looking for *
-genprofile /bin/\*:$okperm
+genprofile /usr/bin/\*:$okperm
 runchecktest "SINGLE TAILGLOB (exec)" pass $file
 
 # PASS TEST - looking for **
-genprofile /bi\*\*:$okperm
+genprofile /usr//bi\*\*:$okperm
 runchecktest "DOUBLE TAILGLOB (exec)" pass $file
 
 # PASS TEST - looking for { , }
-genprofile /bin/\{true,false\}:$okperm
+genprofile /usr/bin/\{true,false\}:$okperm
 runchecktest "CURLY BRACES (exec)" pass $file
 
 # PASS TEST - looking for []
-genprofile /bin/\[aeft\]rue:$okperm
+genprofile /usr/bin/\[aeft\]rue:$okperm
 runchecktest "SQUARE BRACES 1 (exec)" pass $file
 
 # PASS TEST - looking for []
-genprofile /bin/\[s-v\]rue:$okperm
+genprofile /usr/bin/\[s-v\]rue:$okperm
 runchecktest "SQUARE BRACES 2 (exec)" pass $file
 
 # PASS TEST - looking for ?
-genprofile /bin/t\?ue:$okperm
+genprofile /usr/bin/t\?ue:$okperm
 runchecktest "QUESTION MARK (exec)" pass $file
 
 # FAIL TEST - looking for *
-genprofile /sbin/\*:$okperm
+genprofile /usr/sbin/\*:$okperm
 runchecktest "SINGLE TAILGLOB (exec, fail)" fail $file
 
 # FAIL TEST - looking for **
-genprofile /sbi\*\*:$okperm
+genprofile /usr/sbi\*\*:$okperm
 runchecktest "DOUBLE TAILGLOB (exec, fail)" fail $file
 
 # FAIL TEST - looking for { , }
-genprofile /bin/\{flase,false\}:$okperm
+genprofile /usr/bin/\{flase,false\}:$okperm
 runchecktest "CURLY BRACES (exec, fail)" fail $file
 
 # FAIL TEST - looking for []
-genprofile /bin/\[aef\]rue:$okperm
+genprofile /usr/bin/\[aef\]rue:$okperm
 runchecktest "SQUARE BRACES 1 (exec, fail)" fail $file
 
 # FAIL TEST - looking for []
-genprofile /bin/\[u-x\]rue:$okperm
+genprofile /usr/bin/\[u-x\]rue:$okperm
 runchecktest "SQUARE BRACES 2 (exec, fail)" fail $file
 
 # FAIL TEST - looking for ?
-genprofile /bin/b\?ue:$okperm
+genprofile /usr/bin/b\?ue:$okperm
 runchecktest "QUESTION MARK (exec, fail)" fail $file
